Spigot-API-Patches/0218-Create-HoverEvent-from-ItemStack-Entity.patch

```diff
@@ -0,0 +1,110 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ysl3000 <yannicklamprecht@live.de>
+Date: Mon, 6 Jul 2020 22:17:37 +0200
+Subject: [PATCH] Create-HoverEvent-from-ItemStack-Entity
+
+
+diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
+index 4e92ce18227c8f650fe5c6f4a670e6f56d629872..bac4c3802a0babd4dd672df82edfedd9386056ec 100644
+--- a/src/main/java/org/bukkit/Bukkit.java
++++ b/src/main/java/org/bukkit/Bukkit.java
+@@ -13,6 +13,7 @@ import java.util.Set;
+ import java.util.UUID;
+ import java.util.function.Consumer;
+ import java.util.logging.Logger;
++import net.md_5.bungee.api.chat.BaseComponent; // Paper
+ import org.bukkit.Warning.WarningState;
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
+@@ -1759,6 +1760,41 @@ public final class Bukkit {
+     public static com.destroystokyo.paper.entity.ai.MobGoals getMobGoals() {
+         return server.getMobGoals();
+     }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that ItemStack for displaying.
++     *
++     * @param itemStack ItemStack to create the HoverEvent for
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that ItemStack
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull ItemStack itemStack) {
++        return server.hoverContentOf(itemStack);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable String customName) {
++        return server.hoverContentOf(entity, customName);
++    }
++
++    /**
++     * Creates a {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity} for displaying.
++     *
++     * @param entity     Entity to create the HoverEvent for
++     * @param customName a custom name that should be displayed, if not passed entity name will be displayed
++     * @return the {@link net.md_5.bungee.api.chat.hover.content.Content} of that {@link Entity}
++     */
++    @NotNull
++    public static net.md_5.bungee.api.chat.hover.content.Content hoverContentOf(@NotNull Entity entity, @Nullable BaseComponent customName) {
```

Comment:
Shouldn't this accept a `BaseComponent[]` as do other component APIs?

Reply:
Not everything takes an array. Some things require a singular component. You can wrap your array of components in a singular one like so:
```new TextComponent(components)```